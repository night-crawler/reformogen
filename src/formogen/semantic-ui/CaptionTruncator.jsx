import React from 'react';
import PropTypes from 'prop-types';


CaptionTruncatorComponent.displayName = 'CaptionTruncator';
CaptionTruncatorComponent.propTypes = {
  caption: PropTypes.string,
  width: PropTypes.number,
};
export function CaptionTruncatorComponent({ caption, width = 0 }) {
  const inlineStyles = {
    display: 'inline-block',
    maxWidth: width,
    textShadow: '0 0 1px white',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',
    overflow: 'hidden',
  };

  return <span style={ inlineStyles }>{ caption }</span>;
}
