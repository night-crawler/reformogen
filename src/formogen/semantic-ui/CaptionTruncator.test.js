import { CaptionTruncator } from './CaptionTruncator';


it('<TextField />', () => {
  const wrapper = shallow(
    <CaptionTruncator
      caption='1'
      width={ 2 }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
