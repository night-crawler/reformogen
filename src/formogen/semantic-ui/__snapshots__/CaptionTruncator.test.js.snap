// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CaptionTruncator
    caption="1"
    width={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "1",
      "style": Object {
        "display": "inline-block",
        "maxWidth": 2,
        "overflow": "hidden",
        "textOverflow": "ellipsis",
        "textShadow": "0 0 1px white",
        "whiteSpace": "nowrap",
      },
    },
    "ref": null,
    "rendered": "1",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "1",
        "style": Object {
          "display": "inline-block",
          "maxWidth": 2,
          "overflow": "hidden",
          "textOverflow": "ellipsis",
          "textShadow": "0 0 1px white",
          "whiteSpace": "nowrap",
        },
      },
      "ref": null,
      "rendered": "1",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
